graylog2-server/src/main/java/org/graylog2/filters/ExtractorFilter.java

```diff
@@ -75,17 +76,22 @@ public boolean filter(Message msg) {
                 public List<Extractor> call() throws Exception {
                     LOG.debug("Re-loading extractors for input <{}> into cache.", inputId);
 
-                    Input input = inputService.find(inputId);
+                    try {
+                        Input input = inputService.find(inputId);
 
-                    List<Extractor> sorted = Lists.newArrayList(inputService.getExtractors(input));
+                        List<Extractor> sorted = Lists.newArrayList(inputService.getExtractors(input));
 
-                    Collections.sort(sorted, new Comparator<Extractor>() {
-                        public int compare(Extractor e1, Extractor e2) {
-                            return e1.getOrder().intValue() - e2.getOrder().intValue();
-                        }
-                    });
+                        Collections.sort(sorted, new Comparator<Extractor>() {
```

Comment:
Thanks. I agree that would be nicer. I will do that on master only to keep the changes in 1.0 small.